components/gitpod-db/src/typeorm/workspace-cluster-db-impl.ts

```diff
@@ -29,7 +29,10 @@ export class WorkspaceClusterDBImpl implements WorkspaceClusterDB {
 
     async save(cluster: WorkspaceCluster): Promise<void> {
         const repo = await this.getRepo();
-        await repo.save(cluster);
+        await repo.save({
+            ...cluster,
+            deleted: false,
```

Comment:
But how do I mark a WorkspaceCluster as deleted, then, if I cannot store that flag to DB? :shrug:

Reply:
In this case, `save` is only used for creates or updates. Deregister uses `this.clusterDB.deleteByName(req.name);` to delete.

The only other alternative is we expose the `deleted` property in the `WorkspaceCluster` object in protocol